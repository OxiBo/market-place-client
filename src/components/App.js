import React, { Component } from "react";
// import { Link } from "react-router-dom";
import styled, { ThemeProvider, createGlobalStyle } from "styled-components";
// import market from "../styles/images/market.jpeg";
// import { instanceOf } from "prop-types";
import { withCookies } from "react-cookie";
// import Test from "./Test"
// import Login from "./Login"
// import logo from "../logo.svg";
import Header from "./Header";
// import "../App.css";

export const theme = {
  black: "#393939",
  maxWidth: "700px",
  fontColor: "black",
  lightGrey: "#E1E1E1",
  purpleLight: "#D3B7CE",
  purple: "purple",
  sellerColor: "rgb(172, 97, 158)",
  purpleUnderline: "rgb(172, 97, 158)",
  innerContainerBackground: "rgb(225, 234, 250)", //"#D0F0E1"
  toastBackground: "rgba(238, 224, 235, 0.9)"
};

const GlobalStyle = createGlobalStyle`

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
font-size: 15px;

}
*, *:before, *:after{
  box-sizing: inherit;
}
html,
body {
  background: url(${"./tq6XCf.jpg"}) no-repeat center fixed;
  background-size: cover;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
    font-size: 10px;
    h1, h2{
      text-align: center;
    }
}
`;

const Main = styled.div`
  color: ${(props) => props.theme.black};
`;

const Inner = styled.div`
  max-width: ${(props) => props.theme.maxWidth};
  margin: 0 auto;
  padding: 2rem;
  font-size: 1.5rem;
`;

class App extends Component {
  componentDidMount() {
    // const { cookies } = this.props;
    // this.setState({ authToken: cookies.get("token") });
    // console.log(this.props);
  }
  render() {
    // const { cookies } = this.props;
    // const authToken = cookies.get("token");
    // const userName = cookies.get("name")

    return (
      <ThemeProvider theme={theme}>
        <Main>
          <GlobalStyle />
          <Header />
          <Inner>{this.props.children}</Inner>
        </Main>
      </ThemeProvider>
    );
  }
}

export default withCookies(App);
