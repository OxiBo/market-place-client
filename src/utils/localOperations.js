import gql from "graphql-tag";

export const CART_OPEN_QUERY = gql`
  query {
    cartOpen @client
  }
`;


export const TOGGLE_CART_MUTATION = gql`
mutation TODDLE_CART_MUTATION{
    toggleCart @client{
        cartOpen
    }
}`