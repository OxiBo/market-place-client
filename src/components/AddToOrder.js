import React from "react";
import { Mutation } from "react-apollo";
import ItemButton from "./styled/ItemButton";
import { ADD_TO_CART_MUTATION, CART_ITEMS_QUERY } from "../utils/serverOperations";

const AddToOrder = (props) => {
  const { id } = props;
  return (
    <Mutation
      mutation={ADD_TO_CART_MUTATION}
      variables={{ id }}
      refetchQueries={[{query: CART_ITEMS_QUERY}]}
      onCompleted={(data) => console.log(data)}
    >
      {(addToOrder, { data, error, loading }) => {
        {
          /* if (error) console.log(error.message); */
        }
        return <ItemButton disabled={loading} onClick={addToOrder}>Add{loading && "ing"} To Cart</ItemButton>;
      }}
    </Mutation>
  );
};

export default AddToOrder;
