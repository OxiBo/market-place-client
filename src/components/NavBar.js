// TODO - check props and /login (bug - offer to logout if logged in user visit the page or logout automatically?  )

import React, { Component } from "react";
import { Link, withRouter } from "react-router-dom";
import { Mutation, Query } from "react-apollo";
import styled from "styled-components";
import { withCookies, Cookies } from "react-cookie";
import CartCount from "./CartCount";
import { TOGGLE_CART_MUTATION } from "../utils/localOperations";
import { CART_ITEMS_QUERY } from "../utils/serverOperations";
import removeAuthCookies from "../utils/removeAuthCookies";

const Navbar = styled.nav`
  background-color: ${(props) => props.theme.purpleLight};
  min-height: 1rem;
  font-size: calc(10px + 2vmin);
  color: white;
  margin: 0 0 3rem 0;
  width: 100%;
  ul.navbar {
    display: flex;
    justify-content: space-around;

    li {
      display: flex;
      align-items: center;
      justify-content: center;
      font-weight: bold;
      text-transform: uppercase;
      position: relative;
      &:before {
        content: "";
        width: 2px;
        background: ${(props) => props.theme.lightGrey};
        height: 100%;
        /* left: 0; */
        /* position: absolute; */
        transform: skew(-20deg);
        /* top: 0;
        bottom: 0; */
      }
      &:after {
        height: 2px;

        content: "";
        width: 0;

        transform: translateX(-50%);
      }
    }
    a {
      text-decoration: none;
      padding: 0.9rem;
      color: white !important;
      cursor: pointer;
      &:before {
        content: "";
        width: 2px;
        background: ${(props) => props.theme.lightGrey};
        height: 100%;
        left: 0;
        position: absolute;
        transform: skew(-20deg);
        top: 0;
        bottom: 0;
      }
      &:after {
        height: 4px;
        background: ${(props) => props.theme.purpleUnderline};
        content: "";
        width: 0;
        position: absolute;
        transform: translateX(-50%);
        transition: width 0.4s;
        transition-timing-function: cubic-bezier(1, -0.65, 0, 2.31);
        left: 50%;
        margin-top: 2.5rem;
      }
      &:hover,
      &:focus {
        outline: none;
        &:after {
          width: calc(90% - 30px);
        }
      }
    }
  }
`;

const Brand = styled.h2`
  margin-right: auto;
  padding: 0.9rem;
  transform: rotate(-20deg);
`;

class NavBar extends Component {
  render() {
    // console.log(this.props.allCookies)
    // TODO - check out getAll property
    const { cookies } = this.props;
    const authToken = cookies.get("token");
    const userName = cookies.get("name");
    const userId = cookies.get("id");
    const type = cookies.get("type");
    // console.log(this.props.cookies);
    // console.log(this.props);
    // console.log(authToken)
    return (
      <Navbar>
        <ul className="navbar" data-test="nav">
          <Brand>BRAND</Brand>
          <li>
            <Link to="/">Home</Link>
          </li>
          {authToken ? (
            <>
              <li>
                <Link to={`/sellers`}>Sellers</Link>
              </li>

              {type === "SELLER" && (
                <li>
                  <Link to={`/sell`}>sell</Link>
                </li>
              )}
              {type === "BUYER" && (
                <li>
                  <Link to={`/orders`}>My Orders</Link>
                </li>
              )}
              <li>
                {type === "BUYER" ? (
                  <Link to={`/buyer-profile/${userId}`}>
                    {userName}'s profile
                  </Link>
                ) : (
                  <Link to={`/seller-profile/${userId}`}>
                    {userName}'s profile
                  </Link>
                )}
              </li>
              {type === "BUYER" && (
                <li>
                  <Mutation mutation={TOGGLE_CART_MUTATION}>
                    {(toggleCart) => (
                      <a onClick={toggleCart}>
                        My Cart{" "}
                        <Query query={CART_ITEMS_QUERY}>
                          {({ data, error, loading }) => {
                            if (data) {
                              return <CartCount count={data} />;
                            }

                            return null;
                          }}
                        </Query>
                      </a>
                    )}
                  </Mutation>
                </li>
              )}
              <li>
                <a
                  data-test="logout"
                  href=""
                  className=""
                  onClick={() => {
                    removeAuthCookies(cookies);

                    this.props.history.push(`/`);
                  }}
                >
                  Logout
                </a>
              </li>
            </>
          ) : (
            <li>
              <Link to="/login" className="" data-test="login">
                Login
              </Link>
            </li>
          )}
        </ul>
      </Navbar>
    );
  }
}

export default withCookies(withRouter(NavBar));
export { NavBar };
