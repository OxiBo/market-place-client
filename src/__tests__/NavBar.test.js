import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import { act } from "react-dom/test-utils";
import { ApolloConsumer } from "react-apollo";
import { MemoryRouter, withRouter } from "react-router-dom"; // https://html.developreference.com/article/13543175/React+Router+Invariant+Violation+test+fails+with+%3CLink%3E+outside+%3CRouter%3E+%5Bduplicate%5D
// why MemoryRouter - https://github.com/enzymejs/enzyme/issues/1112#issuecomment-447643271
import { NavBar } from "../components/NavBar";
// import { CURRENT_USER_QUERY } from "../components/User";
// import { MockedProvider } from "react-apollo/test-utils";
import { MockedProvider } from "@apollo/react-testing";
 // mocking withCookies - https://github.com/reactivestack/cookies/blob/master/packages/react-cookie/src/__tests__/withCookies-test.js
import { CookiesProvider, withCookies, Cookies } from "react-cookie";

import { CART_ITEMS_QUERY } from "../utils/serverOperations";
import { CART_OPEN_QUERY } from "../utils/localOperations";

// mocking resolvers for queries and mutations with @client directive - https://github.com/apollographql/react-apollo/issues/3316#issuecomment-518948531
const resolvers = {
  defaults: {
    cartOpen: false,
  },
};

const fakeCurrentOrder = {
  id: "abc123",
  items: [
    {
      count: 1,
      id: "abc098",
      image: null,
      name: "test jest",
      price: 2222222,
      __typename: "OrderItem",
      product: {
        id: "prod_cke0uyoym023q0765s7qaqm9k",
        image: null,
        name: "test jest",
        price: 2222222,
        __typename: "Product",
      },
    },
    {
      count: 5,
      id: "qwe098",
      image: null,
      name: "test jest 2",
      price: 1111111,
      __typename: "OrderItem",
      product: {
        id: "prod2_cke0uyoym023q0765s7qaqm9k",
        image: null,
        name: "test jest 2",
        price: 1111111,
        __typename: "Product",
      },
    },
  ],
  __typename: "Order",
};
const signedInMocksWithCartItems = [
  {
    request: { query: CART_ITEMS_QUERY },
    result: {
      data: {
        myCurrentOrder: fakeCurrentOrder,
      },
    },
  },
  {
    request: { query: CART_OPEN_QUERY },
    result: {
      data: {
        cartOpen: false,
      },
    },
  },
];

// const originalError = console.error;

// beforeAll(() => {
//   console.error = jest.fn();
// });

// afterAll(() => {
//   console.error = originalError;
// });

describe("<NavBar />", () => {
  it("renders component without being logged in and match snapshot", async () => {
    const cookies = new Cookies();

    // mocking withCookies - https://github.com/reactivestack/cookies/blob/master/packages/react-cookie/src/__tests__/withCookies-test.js
    const Component = withCookies(NavBar);

    const wrapper = mount(
      <CookiesProvider cookies={cookies}>
        <MemoryRouter>
          <MockedProvider resolvers={resolvers}>
            <Component />
          </MockedProvider>
        </MemoryRouter>
      </CookiesProvider>
    );
    // console.log(wrapper.debug())

    await act(async () => wait());
    wrapper.update();
    const navBar = wrapper.find('ul[data-test="nav"]');
    // console.log(navBar.debug());
    expect(navBar.children().length).toBe(3);

    expect(toJSON(navBar)).toMatchSnapshot();
  });

  it("renders component with buyer logged in and match snapshot", async () => {
    const cookies = new Cookies();
    cookies.set("token", "token_lskdjflskdjflskdjflskdf");
    cookies.set("type", "BUYER");
    cookies.set("name", "Test");
    cookies.set("id", "lskdjflskdjflskdjflskdf");

    const Component = withCookies(NavBar);

    const wrapper = mount(
      <CookiesProvider cookies={cookies}>
        <MemoryRouter>
          <MockedProvider
            mocks={signedInMocksWithCartItems}
            resolvers={resolvers}
          >
            <Component />
          </MockedProvider>
        </MemoryRouter>
      </CookiesProvider>
    );
    // console.log(wrapper.debug())

    await act(async () => wait());
    wrapper.update();
    const navBar = wrapper.find('ul[data-test="nav"]');

    expect(navBar.children().length).toBe(7);
    expect(navBar.text()).toContain("Logout");
    expect(toJSON(navBar)).toMatchSnapshot();
  });

  it("renders component with seller logged in and match snapshot", async () => {
    const cookies = new Cookies();
    cookies.set("token", "token_lskdjflskdjflskdjflskdf");
    cookies.set("type", "SELLER");
    cookies.set("name", "Seller");
    cookies.set("id", "lskdjflskdjflskdjflskdf");

    const Component = withCookies(NavBar);

    const wrapper = mount(
      <CookiesProvider cookies={cookies}>
        <MemoryRouter>
          <MockedProvider
            mocks={signedInMocksWithCartItems}
            resolvers={resolvers}
          >
            <Component />
          </MockedProvider>
        </MemoryRouter>
      </CookiesProvider>
    );
    // console.log(wrapper.debug())

    await act(async () => wait());
    wrapper.update();
    const navBar = wrapper.find('ul[data-test="nav"]');
    // console.log(navBar.debug());
    expect(navBar.children().length).toBe(6);
    expect(toJSON(navBar)).toMatchSnapshot();
  });

  it("renders the amount of items in buyer's cart", async () => {
    const Component = withCookies(NavBar);
    const cookies = new Cookies();
    cookies.set("token", "token_lskdjflskdjflskdjflskdf");
    cookies.set("type", "BUYER");
    cookies.set("name", "Test");
    cookies.set("id", "lskdjflskdjflskdjflskdf");
    const wrapper = mount(
      <CookiesProvider cookies={cookies}>
        <MemoryRouter>
          <MockedProvider
            mocks={signedInMocksWithCartItems}
            resolvers={resolvers}
          >
            <Component />
          </MockedProvider>
        </MemoryRouter>
      </CookiesProvider>
    );
    // console.log(wrapper.debug())

    await act(async () => wait());
    wrapper.update();

    const nav = wrapper.find('ul[data-test="nav"]');
    // console.log(nav);
    const count = nav.find("span.count");
    // console.log(count.text());
    expect(Number(count.text())).toEqual(
      fakeCurrentOrder.items.reduce((sum, item) => sum + item.count, 0)
    );

    // expect(toJSON(count)).toMatchSnapshot();
    // expect(toJSON(nav)).toMatchSnapshot();
  });
  it("signs user out when the logout button is clicked", async () => {
    const cookies = new Cookies();
    cookies.set("token", "token_lskdjflskdjflskdjflskdf");
    cookies.set("type", "BUYER");
    cookies.set("name", "Test");
    cookies.set("id", "lskdjflskdjflskdjflskdf");

    const Component = withRouter(withCookies(NavBar));
    let apolloClient;
    const wrapper = mount(
      <CookiesProvider cookies={cookies}>
        <MemoryRouter>
          <MockedProvider
            mocks={signedInMocksWithCartItems}
            resolvers={resolvers}
          >
            <ApolloConsumer>
              {(client) => {
                apolloClient = client;
                return <Component />;
              }}
            </ApolloConsumer>
          </MockedProvider>
        </MemoryRouter>
      </CookiesProvider>
    );
    await act(async () => wait());
    wrapper.update();
    const logoutButton = wrapper.find('a[data-test="logout"]');
    // console.log(logoutButton.debug());
    logoutButton.simulate("click");
    await act(async () => wait());
    const loginButton = wrapper.find('a[data-test="login"]');
    console.log(loginButton.debug());
    expect(loginButton.text()).toContain("Login");
  });
});
