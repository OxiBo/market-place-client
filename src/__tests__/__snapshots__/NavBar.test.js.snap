// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders component with buyer logged in and match snapshot 1`] = `
<ul
  className="navbar"
  data-test="nav"
>
  <styled.h2>
    <h2
      className="sc-AxhCb iLwveu"
    >
      BRAND
    </h2>
  </styled.h2>
  <li>
    <Link
      to="/"
    >
      <LinkAnchor
        href="/"
        navigate={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Link
      to="/sellers"
    >
      <LinkAnchor
        href="/sellers"
        navigate={[Function]}
      >
        <a
          href="/sellers"
          onClick={[Function]}
        >
          Sellers
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Link
      to="/orders"
    >
      <LinkAnchor
        href="/orders"
        navigate={[Function]}
      >
        <a
          href="/orders"
          onClick={[Function]}
        >
          My Orders
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Link
      to="/buyer-profile/lskdjflskdjflskdjflskdf"
    >
      <LinkAnchor
        href="/buyer-profile/lskdjflskdjflskdjflskdf"
        navigate={[Function]}
      >
        <a
          href="/buyer-profile/lskdjflskdjflskdjflskdf"
          onClick={[Function]}
        >
          Logged in as 
          Test
           
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Mutation
      mutation={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "TODDLE_CART_MUTATION",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [
                      Object {
                        "arguments": Array [],
                        "kind": "Directive",
                        "name": Object {
                          "kind": "Name",
                          "value": "client",
                        },
                      },
                    ],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "toggleCart",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "cartOpen",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 80,
            "start": 0,
          },
        }
      }
    >
      <a
        onClick={[Function]}
      >
        My Cart
         
        <Query
          query={
            Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": Object {
                    "kind": "Name",
                    "value": "CART_ITEMS_QUERY",
                  },
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "myCurrentOrder",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "items",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "count",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "name",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "image",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "price",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "product",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "name",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "price",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "image",
                                          },
                                          "selectionSet": undefined,
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 245,
                "start": 0,
              },
            }
          }
        >
          <CartCount
            count={
              Object {
                "myCurrentOrder": Object {
                  "__typename": "Order",
                  "id": "abc123",
                  "items": Array [
                    Object {
                      "__typename": "OrderItem",
                      "count": 1,
                      "id": "abc098",
                      "image": null,
                      "name": "test jest",
                      "price": 2222222,
                      "product": Object {
                        "__typename": "Product",
                        "id": "prod_cke0uyoym023q0765s7qaqm9k",
                        "image": null,
                        "name": "test jest",
                        "price": 2222222,
                      },
                    },
                    Object {
                      "__typename": "OrderItem",
                      "count": 5,
                      "id": "qwe098",
                      "image": null,
                      "name": "test jest 2",
                      "price": 1111111,
                      "product": Object {
                        "__typename": "Product",
                        "id": "prod2_cke0uyoym023q0765s7qaqm9k",
                        "image": null,
                        "name": "test jest 2",
                        "price": 1111111,
                      },
                    },
                  ],
                },
              }
            }
          >
            <Query
              query={
                Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": Object {
                        "kind": "Name",
                        "value": "CART_ITEMS_QUERY",
                      },
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "myCurrentOrder",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "id",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "items",
                                  },
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "id",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "count",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "name",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "image",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "price",
                                        },
                                        "selectionSet": undefined,
                                      },
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "product",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "id",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "name",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "price",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "image",
                                              },
                                              "selectionSet": undefined,
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 245,
                    "start": 0,
                  },
                }
              }
            >
              <styled.span>
                <span
                  className="sc-AxjAm lmmvdt"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component="div"
                  >
                    <div>
                      <CSSTransition
                        className="count"
                        classNames="count"
                        in={true}
                        key=".$6"
                        onExited={[Function]}
                        timeout={
                          Object {
                            "enter": 4000,
                            "exit": 4000,
                          }
                        }
                        unMountOnExit={true}
                      >
                        <Transition
                          appear={false}
                          className="count"
                          enter={true}
                          exit={true}
                          in={true}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={
                            Object {
                              "enter": 4000,
                              "exit": 4000,
                            }
                          }
                          unMountOnExit={true}
                          unmountOnExit={false}
                        >
                          <styled.span
                            className="count"
                            unMountOnExit={true}
                          >
                            <span
                              className="sc-AxirZ egWJIS count"
                            >
                              6
                            </span>
                          </styled.span>
                        </Transition>
                      </CSSTransition>
                    </div>
                  </TransitionGroup>
                </span>
              </styled.span>
            </Query>
          </CartCount>
        </Query>
      </a>
    </Mutation>
  </li>
  <li>
    <a
      className=""
      data-test="logout"
      href=""
      onClick={[Function]}
    >
      Logout
    </a>
  </li>
</ul>
`;

exports[`<NavBar /> renders component with seller logged in and match snapshot 1`] = `
<ul
  className="navbar"
  data-test="nav"
>
  <styled.h2>
    <h2
      className="sc-AxhCb iLwveu"
    >
      BRAND
    </h2>
  </styled.h2>
  <li>
    <Link
      to="/"
    >
      <LinkAnchor
        href="/"
        navigate={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Link
      to="/sellers"
    >
      <LinkAnchor
        href="/sellers"
        navigate={[Function]}
      >
        <a
          href="/sellers"
          onClick={[Function]}
        >
          Sellers
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Link
      to="/sell"
    >
      <LinkAnchor
        href="/sell"
        navigate={[Function]}
      >
        <a
          href="/sell"
          onClick={[Function]}
        >
          sell
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Link
      to="/seller-profile/lskdjflskdjflskdjflskdf"
    >
      <LinkAnchor
        href="/seller-profile/lskdjflskdjflskdjflskdf"
        navigate={[Function]}
      >
        <a
          href="/seller-profile/lskdjflskdjflskdjflskdf"
          onClick={[Function]}
        >
          Logged in as 
          Seller
           
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <a
      className=""
      data-test="logout"
      href=""
      onClick={[Function]}
    >
      Logout
    </a>
  </li>
</ul>
`;

exports[`<NavBar /> renders component without being logged in and match snapshot 1`] = `
<ul
  className="navbar"
  data-test="nav"
>
  <styled.h2>
    <h2
      className="sc-AxhCb iLwveu"
    >
      BRAND
    </h2>
  </styled.h2>
  <li>
    <Link
      to="/"
    >
      <LinkAnchor
        href="/"
        navigate={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li>
    <Link
      className=""
      data-test="login"
      to="/login"
    >
      <LinkAnchor
        className=""
        data-test="login"
        href="/login"
        navigate={[Function]}
      >
        <a
          className=""
          data-test="login"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </LinkAnchor>
    </Link>
  </li>
</ul>
`;
